### PR TITLE
provider: Support default tags (resources aws_e* 2)

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Reference: https://github.com/hashicorp/terraform-provider-aws/issues/7926

Output from acceptance testing: Will be handled via CI, review, and daily acceptance testing

### CODE REVIEW

File: aws/resource_aws_ec2_traffic_mirror_filter.go [hunk 1]
```diff
@@ -41,22 +43,25 @@ func resourceAwsEc2TrafficMirrorFilter() *schema.Resource {
...
 	input := &ec2.CreateTrafficMirrorFilterInput{}
 
 	if description, ok := d.GetOk("description"); ok {
 		input.Description = aws.String(description.(string))
 	}
 
-	if v, ok := d.GetOk("tags"); ok {
-		input.TagSpecifications = ec2TagSpecificationsFromMap(v.(map[string]interface{}), ec2.ResourceTypeTrafficMirrorFilter)
+	if len(tags) > 0 {
+		input.TagSpecifications = ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeTrafficMirrorFilter)
```
Comment (YakDriver) [inline aws/resource_aws_ec2_traffic_mirror_filter.go:64]: Ignore is not needed?

File: aws/resource_aws_ec2_traffic_mirror_session.go [hunk 2]
```diff
@@ -92,8 +97,8 @@ func resourceAwsEc2TrafficMirrorSessionCreate(d *schema.ResourceData, meta inter
 		input.VirtualNetworkId = aws.Int64(int64(v.(int)))
 	}
 
-	if v, ok := d.GetOk("tags"); ok {
-		input.TagSpecifications = ec2TagSpecificationsFromMap(v.(map[string]interface{}), ec2.ResourceTypeTrafficMirrorSession)
+	if len(tags) > 0 {
+		input.TagSpecifications = ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeTrafficMirrorSession)
```
Comment (YakDriver) [inline aws/resource_aws_ec2_traffic_mirror_session.go:101]: Ignore?

File: aws/resource_aws_ec2_traffic_mirror_target.go [hunk 3]
```diff
@@ -74,8 +81,8 @@ func resourceAwsEc2TrafficMirrorTargetCreate(d *schema.ResourceData, meta interf
 		input.NetworkLoadBalancerArn = aws.String(v.(string))
 	}
 
-	if v, ok := d.GetOk("tags"); ok {
-		input.TagSpecifications = ec2TagSpecificationsFromMap(v.(map[string]interface{}), ec2.ResourceTypeTrafficMirrorTarget)
+	if len(tags) > 0 {
+		input.TagSpecifications = ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeTrafficMirrorTarget)
```
Comment (YakDriver) [inline aws/resource_aws_ec2_traffic_mirror_target.go:85]: Ignore?

File: aws/resource_aws_ec2_transit_gateway.go [hunk 4]
```diff
@@ -124,7 +130,7 @@ func resourceAwsEc2TransitGatewayCreate(d *schema.ResourceData, meta interface{}
 			DnsSupport:                   aws.String(d.Get("dns_support").(string)),
 			VpnEcmpSupport:               aws.String(d.Get("vpn_ecmp_support").(string)),
 		},
-		TagSpecifications: ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2.ResourceTypeTransitGateway),
+		TagSpecifications: ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeTransitGateway),
```
Comment (YakDriver) [inline aws/resource_aws_ec2_transit_gateway.go:133]: Ignore?

File: aws/resource_aws_ec2_transit_gateway_peering_attachment.go [hunk 5]
```diff
@@ -59,7 +64,7 @@ func resourceAwsEc2TransitGatewayPeeringAttachmentCreate(d *schema.ResourceData,
 		PeerAccountId:        aws.String(peerAccountId),
 		PeerRegion:           aws.String(d.Get("peer_region").(string)),
 		PeerTransitGatewayId: aws.String(d.Get("peer_transit_gateway_id").(string)),
-		TagSpecifications:    ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2.ResourceTypeTransitGatewayAttachment),
+		TagSpecifications:    ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeTransitGatewayAttachment),
```
Comment (YakDriver) [inline aws/resource_aws_ec2_transit_gateway_peering_attachment.go:67]: Ignore?

File: aws/resource_aws_ec2_transit_gateway_peering_attachment_accepter.go [hunk 6]
```diff
@@ -66,8 +71,8 @@ func resourceAwsEc2TransitGatewayPeeringAttachmentAccepterCreate(d *schema.Resou
 		return fmt.Errorf("error waiting for EC2 Transit Gateway Peering Attachment (%s) availability: %s", d.Id(), err)
 	}
 
-	if v := d.Get("tags").(map[string]interface{}); len(v) > 0 {
-		if err := keyvaluetags.Ec2CreateTags(conn, d.Id(), v); err != nil {
+	if len(tags) > 0 {
+		if err := keyvaluetags.Ec2CreateTags(conn, d.Id(), tags); err != nil {
```
Comment (YakDriver) [inline aws/resource_aws_ec2_transit_gateway_peering_attachment_accepter.go:75]: Ignore?

File: aws/resource_aws_ec2_transit_gateway_route_table.go [hunk 7]
```diff
@@ -48,10 +51,12 @@ func resourceAwsEc2TransitGatewayRouteTable() *schema.Resource {
 
 func resourceAwsEc2TransitGatewayRouteTableCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(keyvaluetags.New(d.Get("tags").(map[string]interface{})))
 
 	input := &ec2.CreateTransitGatewayRouteTableInput{
 		TransitGatewayId:  aws.String(d.Get("transit_gateway_id").(string)),
-		TagSpecifications: ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2.ResourceTypeTransitGatewayRouteTable),
+		TagSpecifications: ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeTransitGatewayRouteTable),
```
Comment (YakDriver) [inline aws/resource_aws_ec2_transit_gateway_route_table.go:59]: Ignore?

File: aws/resource_aws_ec2_transit_gateway_vpc_attachment.go [hunk 8]
```diff
@@ -96,7 +101,7 @@ func resourceAwsEc2TransitGatewayVpcAttachmentCreate(d *schema.ResourceData, met
 		},
 		SubnetIds:         expandStringSet(d.Get("subnet_ids").(*schema.Set)),
 		TransitGatewayId:  aws.String(transitGatewayID),
-		TagSpecifications: ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2.ResourceTypeTransitGatewayAttachment),
+		TagSpecifications: ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeTransitGatewayAttachment),
```
Comment (YakDriver) [inline aws/resource_aws_ec2_transit_gateway_vpc_attachment.go:104]: Ignore?

File: aws/resource_aws_ec2_transit_gateway_vpc_attachment_accepter.go [hunk 9]
```diff
@@ -90,8 +95,8 @@ func resourceAwsEc2TransitGatewayVpcAttachmentAccepterCreate(d *schema.ResourceD
 		return fmt.Errorf("error waiting for EC2 Transit Gateway VPC Attachment (%s) availability: %s", d.Id(), err)
 	}
 
-	if v := d.Get("tags").(map[string]interface{}); len(v) > 0 {
-		if err := keyvaluetags.Ec2CreateTags(conn, d.Id(), v); err != nil {
+	if len(tags) > 0 {
+		if err := keyvaluetags.Ec2CreateTags(conn, d.Id(), tags); err != nil {
```
Comment (YakDriver) [inline aws/resource_aws_ec2_transit_gateway_vpc_attachment_accepter.go:99]: Ignore?

File: aws/resource_aws_ec2_transit_gateway.go [hunk 10]
```diff
@@ -25,15 +25,18 @@ func resourceAwsEc2TransitGateway() *schema.Resource {
...
-		CustomizeDiff: customdiff.Sequence(
-			customdiff.ForceNewIfChange("default_route_table_association", func(_ context.Context, old, new, meta interface{}) bool {
-				// Only changes from disable to enable for feature_set should force a new resource
-				return old.(string) == ec2.DefaultRouteTableAssociationValueDisable && new.(string) == ec2.DefaultRouteTableAssociationValueEnable
-			}),
-			customdiff.ForceNewIfChange("default_route_table_propagation", func(_ context.Context, old, new, meta interface{}) bool {
-				// Only changes from disable to enable for feature_set should force a new resource
-				return old.(string) == ec2.DefaultRouteTablePropagationValueDisable && new.(string) == ec2.DefaultRouteTablePropagationValueEnable
-			}),
+		CustomizeDiff: customdiff.All(
```
Comment (anGie44) [inline aws/resource_aws_ec2_transit_gateway.go:28]: i'll double-check on this one but the patterns i've seen so far tend to just use `customdiff.Sequence()`, without the wrapped `customdiff.All()` 🤔

**Review by anGie44 — APPROVED** [review]
Just an open comment that'll I'll circle back to, otherwise LGTM

**Pull request merged**